extendr-api/tests/altrep_tests.rs

```diff
@@ -231,7 +232,51 @@ fn test_altstring() {
         let class = Altrep::make_altstring_class::<StringInts>("si", "mypkg");
         let obj = Altrep::from_state_and_class(mystate, class, false);
 
+        assert!(obj.is_altstring());
         assert_eq!(obj.len(), 10);
         assert_eq!(Robj::from(obj), r!(["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]));
     }
 }
+
+#[test]
+#[cfg(use_r_altlist)]
+fn test_altlist() {
+    use extendr_api::AltListImpl;
+    with_r(|| {
+        #[derive(Debug, Clone)]
+        pub struct VecUsize(pub Vec<Option<usize>>);
+
+        // need to make the VecUsize object `.into_robj()`-able
+        #[extendr]
+        impl VecUsize {}
+
+        impl AltrepImpl for VecUsize {
+            fn length(&self) -> usize {
+                self.0.len()
+            }
+        }
+
+        impl AltListImpl for VecUsize {
+            fn elt(&self, index: usize) -> Robj {
+                Self(vec![self.0[index]]).into_robj()
+            }
+        }
+
+        let vu = VecUsize(vec![Some(1), None, Some(10)]);
+
+        let class = Altrep::make_altlist_class::<VecUsize>("li", "mypkg");
+        let obj = Altrep::from_state_and_class(vu, class, false);
+
+        // confirm it is altlist
+        assert!(obj.is_altlist());
+
+        // confirm method is accurate
+        assert_eq!(obj.len(), 3);
+
+        // convert to a list and test the .elt() method
+        let l = List::try_from(obj.into_robj()).unwrap();
+        let li = l.elt(1).unwrap();
```

Comment:
I think it should be just trivially indexable with `[1]`, no? If `L` is of `Lilst` type and it does not support `[]`, it is an oversight.

Reply:
Hm, doing so results in the following: 

Error: cannot take a writable DATAPTR of an ALTLIST [class: li, pkg: mypkg]
Fatal error: unable to initialize the JIT

Reply:
Do you have an idea of what this means?

Reply:
>
    /* Disallow taking the writable `DATAPTR()` of an ALTLIST. This
       check could be moved to `DATAPTR()` to catch more faulty
       usages. */